File: lib/active_admin_bootstrap/extensions/inputs/tom_select_input.rb
require_relative("./select_input")

module ActiveAdmin
  module Inputs
    class TomSelectInput < SelectInput
      include ::ActiveAdminBootstrap::Extensions::Inputs::Base

      # @TODO: allow modifying js configs
      def input_html_options
        super.merge(data: {"aa-tom-select": {}})
      end
    end
  end
end

File: app/javascript/active_admin_bootstrap/lib/inputs/tom_select_input.js
import SelectInput from './select_input'

class TomSelectInput extends SelectInput {
  _defaultInputHTMLOptions () {
    const options = {
      'data-aa-tom-select': {}
    }

    return { ...super._defaultInputHTMLOptions(), ...options }
  }
}

export default TomSelectInput
